### PR TITLE
Permit

### PR DESCRIPTION
**Pull request by MrChico**
fixes permit and adds some tests

### CODE REVIEW

File: contracts/Governance/RadicleToken.sol [hunk 1]
```diff
@@ -120,6 +120,13 @@ contract RadicleToken {
         return DECIMALS;
     }
 
+    /* @notice DOMAIN_SEPARATOR */
+    function DOMAIN_SEPARATOR() public view returns (bytes32) {
+        return keccak256(
+                abi.encode(DOMAIN_TYPEHASH, keccak256(bytes(NAME)), getChainId(), address(this))
+               );
```
Comment (d-xo) [inline contracts/Governance/RadicleToken.sol]: nit: indentation

File: bin/permit [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```
Comment (d-xo) [inline bin/permit:1]: I think we could just include `token` in the `shell.nix`?

Comment (MrChico) [reply bin/permit:1]: I modified it quite heavily to not rely on rpc calls to get the data that is to be signed

File: contracts/Governance/RadicleToken.sol (hunk 1 above)
Comment (MrChico) [reply contracts/Governance/RadicleToken.sol]: how do you want it?

Comment (d-xo) [inline contracts/Governance/RadicleToken.sol]:
```suggestion
        return keccak256(
            abi.encode(DOMAIN_TYPEHASH, keccak256(bytes(NAME)), getChainId(), address(this))
        );
```

**Pull request merged**